### PR TITLE
Auth branch

### PR DESCRIPTION
**Pull request by nayyyhaa**
## In this PR :
-  I've just added private route auth integration + toast + component.
- added a few more products in db.

#### Component files

- [x] [components/Card.js](https://github.com/nayyyhaa/inNotion-care-react/commit/8b4431fd6fd5950b2d7676ce0c706895fffd661b#diff-7916963985f6b54523951aa9bdcdb855db5905bed40fc7775a485986aec0fa5d)
- [x] [components/CartCard.js](https://github.com/nayyyhaa/inNotion-care-react/commit/8b4431fd6fd5950b2d7676ce0c706895fffd661b#diff-052126a8921eb24c8d036b50b456f25f263db9fbc51b55153175215de5afc297)
- [x] [components/CartCard.js](https://github.com/nayyyhaa/inNotion-care-react/commit/8b4431fd6fd5950b2d7676ce0c706895fffd661b#diff-052126a8921eb24c8d036b50b456f25f263db9fbc51b55153175215de5afc297)

 #### Contexts  + Reducers

- [x] [contexts/CartContext.js](https://github.com/nayyyhaa/inNotion-care-react/commit/8b4431fd6fd5950b2d7676ce0c706895fffd661b#diff-278f8b004170b9f9f18d7ec88ac20a13a2947eb5f581b5ba3f6b8734f75db8a7)
- [x] [contexts/ToastContext.js](https://github.com/nayyyhaa/inNotion-care-react/commit/8b4431fd6fd5950b2d7676ce0c706895fffd661b#diff-d9068a4abb970500dbdc01c2c3fa89bfa56ea5c5fa2d80e095c818807dc9290f)
- [x] [contexts/WishlistContext.js](https://github.com/nayyyhaa/inNotion-care-react/commit/8b4431fd6fd5950b2d7676ce0c706895fffd661b#diff-d11079295c460be6c868875c3fc185c0e3e0c181df4a3fa4d1b1e9adc43d7691)
- [x] [toastReducer.js](https://github.com/nayyyhaa/inNotion-care-react/commit/8b4431fd6fd5950b2d7676ce0c706895fffd661b#diff-49f1e0ae1fd946b13bc00b3fe914039a4d66a3ded26cd771b2c613751b1ca434)

#### customHooks + service:
- [x] [custom-hooks/useLogin.js](https://github.com/nayyyhaa/inNotion-care-react/commit/8b4431fd6fd5950b2d7676ce0c706895fffd661b#diff-76ae2ea4f8f82005d5fe82d3b89b1787da4a8378677688dcfb2c9ae467e561df)
- [x] [custom-hooks/useScrollToTop.js](https://github.com/nayyyhaa/inNotion-care-react/commit/8b4431fd6fd5950b2d7676ce0c706895fffd661b#diff-58372974f338889949d11725144fc849b6c3117b73017ad49cb8f2e2f6bd85ad)
- [x] [utils/cartService.js](https://github.com/nayyyhaa/inNotion-care-react/commit/8b4431fd6fd5950b2d7676ce0c706895fffd661b#diff-1d35c00377af07cdf87a1bb5d6929345425d588784e4ed1dbdc247fb333f784b)
- [x] [utils/wishlistService.js](https://github.com/nayyyhaa/inNotion-care-react/commit/8b4431fd6fd5950b2d7676ce0c706895fffd661b#diff-380674ee694ed45f3547f811dc09bed3ad7a2be11244fd3d068ce79bf66db15e)
 
#### files to *ignore*
- [ ] package.json files
- [ ] files with just few import changed
- [ ] CSS files

### Live preview : https://deploy-preview-6--innotion-carev2.netlify.app/

### CODE REVIEW

**Review by abhisheksahani-nova — APPROVED** [review]
Good UI , clean and readable code , seems good to merge

**Comment by abhisheksahani-nova** [comment]
> Here I've used `e` for using preventDefault in form submission.

Ohh okay got it

File: src/contexts/CartContext.js [hunk 1]
```diff
@@ -1,11 +1,86 @@
...
+          type: "SHOW_TOAST",
+          payload: { state: "success", msg: "Added to Cart!" },
+        });
+      } else navigate("/login");
+    } catch (err) {
+      dispatchToast({
+        type: "SHOW_TOAST",
+        payload: { state: "error", msg: "Error in adding product to cart" },
+      });
+      console.log(err);
```
Comment (foolhardy21) [inline src/contexts/CartContext.js]: Now that toast has been added, you may remove the `console.log`

File: src/pages/ProfilePage.js [hunk 2]
```diff
@@ -23,7 +29,14 @@ export const ProfilePage = () => {
               <strong className="m-r-1">Email:</strong>
               {user?.email}
             </li>
-            <button className="btn primary-btn centered-text w-95p m-h-1" onClick={logoutHandler}>
+            <button
+              className="btn primary-btn centered-text w-95p m-h-1"
+              onClick={() => {
```
Comment (foolhardy21) [inline src/pages/ProfilePage.js]: Just a suggestion - you may shift these functions inside a handler. That will look cleaner.

**Review by foolhardy21 — APPROVED** [review]
Nice work. Code looks good.

**Comment by nayyyhaa** [comment]
Thanks @foolhardy21 & @abhisheksahani-nova for reviewing. Merging the PR

**Pull request merged**